ls1intum/Artemis
# Extend modeling plagiarism checks

This PR adds support for checking plagiarism for modeling exercises with the following diagram types:
- [x] Use case diagrams
- [x] Component diagrams
- [x] Deployment diagrams

### Checklist

### Motivation and Context
We would like to be able to check plagiarism for additional diagram types

### Description
This PR adds the corresponding Json parsers and similarity checks for the above mentioned diagram types.
It does **not** change any client files and also does **not** change existing REST endpoints.

### Steps for Testing
1. Create modeling exercises with diagram types use case, component, deployment (instructor)
2. Create several (similar and different) modeling submissions to those modeling exercises
3. Navigate into modeling exercise --> view
4. Click on Check plagiarism and check if the similarity values in the downloaded file of the REST response make sense. 
   * Equal models should have a similarity of 1. 
   * Similarity values are between 0 and 1. Lower values indicate less similarity, higher values indicate more similarity
